packages/server/test/unit/cloud/api/putProtocolArtifact_spec.ts

```diff
@@ -0,0 +1,180 @@
+import mockery from 'mockery'
+import { enable as enableMockery, mockElectron } from '../../../mockery_helper'
+import sinon from 'sinon'
+import chai, { expect } from 'chai'
+import sinonChai from 'sinon-chai'
+import chaiAsPromised from 'chai-as-promised'
+
+import { ReadStream } from 'fs'
+import { StreamActivityMonitor } from '../../../../lib/cloud/upload/StreamActivityMonitor'
+import { uploadStream } from '../../../../lib/cloud/upload/uploadStream'
+import { HttpError } from '../../../../lib/cloud/api/HttpError'
+
+chai.use(chaiAsPromised).use(sinonChai)
+
+describe('putProtocolArtifact', () => {
+  let filePath: string
+  let maxFileSize: number
+  let fileSize: number
+  let mockStreamMonitor
+  let mockReadStream
+  let destinationUrl
+
+  let statStub: sinon.SinonStub
+  let uploadStreamStub: sinon.SinonStub<Parameters<typeof uploadStream>, ReturnType<typeof uploadStream>>
+  let geometricRetryStub: sinon.SinonStub
+
+  let putProtocolArtifact: (artifactPath: string, maxFileSize: number, destinationUrl: string) => Promise<void>
+
+  before(() => {
+    if (global.mockery) {
+      global.mockery.disable()
+    }
+  })
+
+  after(() => {
+    if (global.mockery) {
```

Comment:
`global.mockery` gets set the first time `spec_helper.js` is imported.

This turns the global mockery (that a lot of other specs rely on being implicitly initialized, even if they don't actually import `spec_helper.js`) off before `put_protocol_artifact` top level describe block runs, and turns it back on again afterwards. It doesn't undefine the global mockery, it just disables it and re-enables it.

I'll drop a comment in here to explain the workaround. The alternative was to refactor all other tests that rely on `spec_helper`, either implicitly or explicitly.  I think we can work on refactoring that one case at a time rather than bloating this PR.

Reply:
resolved in d74c5dd170ff15d3ea0125a6ed5e7f22de5d91f7